sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala

```diff
@@ -2333,7 +2333,12 @@ object QueryCompilationErrors {
   }
 
   def cannotModifyValueOfSparkConfigError(key: String): Throwable = {
-    new AnalysisException(s"Cannot modify the value of a Spark config: $key")
+    new AnalysisException(
+      s"""
+      |Cannot modify the value of a Spark config: $key,
+      |please set spark.sql.legacy.setCommandRejectsSparkCoreConfs as 'false' in
+      |order to make change value of Spark config: $key .
```

Comment:
+1 for @HyukjinKwon comment.

Reply:
@HyukjinKwon @dongjoon-hyun  Thanks for reviewing this PR. Yes I got. But until we remove sql legacy configurations. This additional info in ERROR message will be helpful for the users to fix the issue quickly( otherwise they need to find the doc and then resolve this issue while modifying  the value of SparkConf.)

Reply:
The point is that we are trying to avoid hard-wiring the confs because we forget them easily. 
Maybe, we can give an indirect link like migration instead. WDTY?

Reply:
Sure